src/gui/wizard/owncloudadvancedsetuppage.cpp

```diff
@@ -139,11 +125,31 @@ void OwncloudAdvancedSetupPage::initializePage()
     connect(quotaJob, &PropfindJob::result, this, &OwncloudAdvancedSetupPage::slotQuotaRetrieved);
     quotaJob->start();
 
+    QJsonDocument configJson = acc->property("oc_serverConfig").toJsonDocument();
+    bool hasServerConfig = configJson.isObject() && !configJson.object()["default_folders"].toArray().isEmpty();
+    Theme *theme = Theme::instance();
 
-    if (Theme::instance()->wizardSelectiveSyncDefaultNothing()) {
+    _ui.rDefaultFolders->setVisible(hasServerConfig);
+    _ui.rSyncEverything->setVisible(!hasServerConfig);
+    _ui.lSyncEverythingSizeLabel->setVisible(!hasServerConfig);
+    _ui.rSelectiveSync->setVisible(!hasServerConfig);
+    _ui.bSelectiveSync->setVisible(!hasServerConfig);
+    _ui.lSelectiveSyncSizeLabel->setVisible(!hasServerConfig);
+    _ui.rVirtualFileSync->setVisible(!hasServerConfig && Theme::instance()->showVirtualFilesOption() && bestAvailableVfsMode() != Vfs::Off);
```

Comment:
As far as I remember removing the layout from the wSyncStrategy widget was required for making the layout not be broken on some platforms. May need testing.